pandas/_libs/tslib.pyi

```diff
@@ -8,7 +8,7 @@ def format_array_from_datetime(
     values: npt.NDArray[np.int64],
     tz: tzinfo | None = ...,
     format: str | None = ...,
-    na_rep: object = ...,
+    na_rep: str | float = ...,
```

Comment:
`format_array_from_datetime` is only called from 

https://github.com/pandas-dev/pandas/blob/7c208c8907f5ab18f807366c0c5e26ae1dbca299/pandas/core/arrays/datetimes.py#L709-L718

, where `na_rep` can't be `None`